kernel/timehires.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * timehires.c
+ *
+ * based on code from linux/kernel/time/timecounter.c
+ *
+ * Copyright (C) 2018 Petri Mattila <petri.mattila@unipart.io>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ */
+
+#include <linux/time.h>
+#include <linux/export.h>
+
+#include "timehires.h"
+
+
+typedef union {
+	u64 D;
+	struct {
+#ifdef __BIG_ENDIAN
+		u32 H, L;
+#else
+		u32 L, H;
+#endif
+	};
+} n64;
```

Comment:
I'm unclear what "n64" is supposed to mean.  Does this need a more descriptive type name?